src/vm.rs

```diff
@@ -670,22 +670,38 @@ impl<T: GuestStagePagingMode> Vm<T, VmStateFinalized> {
             counter_mask: u64,
             start_flags: PmuCounterStartFlags,
             initial_value: u64,
-            _active_vcpu: &mut ActiveVmCpu<T>,
+            active_vcpu: &mut ActiveVmCpu<T>,
         ) -> EcallResult<u64> {
-            let _ = PmuInfo::get()?;
+            let counter_mask = active_vcpu
+                .pmu()
+                .get_startable_counter_range(counter_index, counter_mask)?;
             pmu::start_counters(counter_index, counter_mask, start_flags, initial_value)?;
+            active_vcpu
+                .pmu()
+                .update_started_counters(counter_index, counter_mask);
```

Comment:
Per the chat with @atishp04 and https://github.com/rivosinc/salus/issues/59, the plan is to match the SBI implementation for now, and address when a more strict implementation is rolled out. Will add a comment to reflect it.

Reply:
So there is an inconsistency here then: if we're following the current OpenSBI implementation, then we should still `update_started_counters()` even if we get `AlreadyStarted` since OpenSBI will have started any other startable counters in the mask.

Reply:
Think this went into the next commit -- it's a little difficult to see the entire code associated with the comment, but will pull in in the previous commit if it's the case.

```
// Special case "already started" to handle counters that are autostarted following configuration.
            // Examples of such counters include the legacy timer and insret.
            if let Err(e) = result && !matches!(e, SbiError::AlreadyStarted) {
                return Err(EcallError::from(e));
            }
```

Reply:
Ah. Yeah, this should really go in the "PMU counter bookkeeping" commit.